token-erc-20/chaincode-go/chaincode/token_contract.go

```diff
@@ -84,11 +83,104 @@ func (s *SmartContract) Mint(ctx contractapi.TransactionContextInterface, amount
 	totalSupply += amount
 	ctx.GetStub().PutState(totalSupplyKey, []byte(strconv.Itoa(totalSupply)))
 
+	// Emit the Transfer event
+	transferEvent := map[string]string{"from": "0x0", "to": minter, "value": strconv.Itoa(amount)}
```

Comment:
This will make the value a string in the JSON. The value needs to be an integer. You'll need to create a struct that has strings for "from" and "to", and an integer for "value", and then Marshal() it.

Same change should be made throughout.